test/server/cards/01_SOR/units/DisablingFangFighter.spec.ts

```diff
@@ -0,0 +1,58 @@
+describe('Disabling Fang Fighter', function() {
+    integration(function(contextRef) {
+        describe('Disabling Fang Fighter\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['disabling-fang-fighter'],
+                        groundArena: [{ card: 'pyke-sentinel', upgrades: ['frozen-in-carbonite'] }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'luke-skywalker#jedi-knight', upgrades: ['lukes-lightsaber'] }],
+                    }
+                });
+            });
+
+            it('can defeat an upgrade on a friendly unit', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.disablingFangFighter);
+                expect(context.player1).toBeAbleToSelectExactly([context.frozenInCarbonite, context.lukesLightsaber]);
+                expect(context.player1).toHavePassAbilityButton();
+
+                context.player1.clickCard(context.frozenInCarbonite);
+                expect(context.pykeSentinel.isUpgraded()).toBe(false);
+                expect(context.lukeSkywalkerJediKnight.isUpgraded()).toBe(true);
+                expect(context.frozenInCarbonite).toBeInZone('discard');
+            });
+
+            it('can defeat an upgrade on a enemy unit', function () {
```

Comment:
Thanks for the good test cases! Unfortunately, we have this practical problem where the jasmine tests are very slow and heavy and we have finite Github Actions time in the free tier that we have to conserve. So we try to consolidate our test cases as much as we can.

In this case we generally assume that if a target appears in the target list, we can assume that it will work unless there's a special case. Same for the pass button, so long as we check that it appears we assume it works. So we can remove the bottom two test cases here.

Reply:
Understood and fixed